lib/core-integration/src/Test/Integration/Framework/Request.hs

```diff
@@ -176,23 +183,38 @@ request ctx (verb, path) reqHeaders body = do
                 NonJson x -> RequestBodyLBS x
                 Empty -> mempty
 
-    handleResponse res = case responseStatus res of
-        s | s < status500 -> either
-            (\err -> (s, Left $ DecodeFailure $ BL8.pack $ err <> ": " <> show res))
-            ((s,) . Right)
-            (Aeson.eitherDecode $ responseBody res)
-
-        -- TODO: decode API error responses into ClientError
-        s -> (s, Left $ decodeFailure res)
+    handleResponse res = do
+        chunks <- brConsume (responseBody res)
+        pure $ case responseStatus res of
+            s | s < status500 ->
+                case decodeOne chunks <|> decodeStream chunks of
+                    Nothing -> decodeFailure s chunks
+                    Just a  -> (s, Right a)
+            -- TODO: decode API error responses into ClientError
+            s -> decodeFailure s chunks
+
+    -- NOTE
+    -- The API now returns responses either as one chunk, or, as a list of chunks
+    -- in case it's streaming responses. In practice, a smart client would process
+    -- chunk one by one, but within our integration framework, it is easier to
+    -- consider streams as a list.
+    --
+    -- Therefore, in case where the response is stream, we simply decode each
+    -- chunk as a generic JSON, re-encode them as a list (trimming the '\n'
+    -- separator), and decoding them as a JSON list.
+    decodeOne = Aeson.decodeStrict . mconcat
+    decodeStream chunks = do
+        xs <- traverse (Aeson.decodeStrict @Aeson.Value . B8.init) chunks
```

Comment:
I don't think you can necessarily assume that the chunks are split nicely at the newline.
The OS might interrupt the socket read early, and you will have a line in two pieces, neither of which are well-formed json.

This streaming change is worthy of its own PR, where the details can be sorted out.

I like the idea of streaming in general, because it removes one case of unbounded memory usage.

But on the other hand, if you have 1 million addresses, that would be perhaps 80MB of encoded json, as an indication of size. So the memory cost of not streaming is smallish. Without streaming there is a delay while the server is reading the list from sqlite, but unless the clients also have stream processing, so what?

Reply:
@rvl actually.. there's one test failing because of this which shows that sometimes, there are chunks that only contains half of an address (it fails on the "import 15.000 addresses and list them" scenario, I haven't noticed this below 100 addresses). 

So yeah, this require some additional thoughts maybe. I'll split the streaming approach into a separate PR.